app/src/main/java/org/oppia/android/app/home/recentlyplayed/RecentlyPlayedFragmentPresenter.kt

```diff
@@ -39,189 +35,48 @@ class RecentlyPlayedFragmentPresenter @Inject constructor(
-      entityType,
-      fragment as PromotedStoryClickListener,
-      index,
-      resourceHandler,
-      translationController
-    )
-  }
-
-  private fun addOlderStoryListSection(olderPlayedStoryList: List<PromotedStory>) {
-    val showDivider = itemList.isNotEmpty()
-    val olderSectionTitleViewModel =
-      SectionTitleViewModel(
-        resourceHandler.getStringInLocale(R.string.ongoing_story_last_month),
-        showDivider
-      )
-    itemList.add(olderSectionTitleViewModel)
-    olderPlayedStoryList.forEachIndexed { index, promotedStory ->
-      val ongoingStoryViewModel = createOngoingStoryViewModel(promotedStory, index)
-      itemList.add(ongoingStoryViewModel)
-    }
-  }
 
-  private fun addRecommendedStoryListSection(suggestedStoryList: List<PromotedStory>) {
-    val showDivider = itemList.isNotEmpty()
-    val recommendedSectionTitleViewModel =
-      SectionTitleViewModel(
-        resourceHandler.getStringInLocale(R.string.recommended_stories),
-        showDivider
-      )
-    itemList.add(recommendedSectionTitleViewModel)
-    suggestedStoryList.forEachIndexed { index, suggestedStory ->
-      val ongoingStoryViewModel = createOngoingStoryViewModel(suggestedStory, index)
-      itemList.add(ongoingStoryViewModel)
-    }
-  }
-
-  private fun getAssumedSuccessfulPromotedActivityList(): LiveData<PromotedActivityList> {
-    return Transformations.map(promotedStoryListSummaryResultLiveData) {
-      when (it) {
-        // If there's an error loading the data, assume the default.
-        is AsyncResult.Failure, is AsyncResult.Pending -> PromotedActivityList.getDefaultInstance()
-        is AsyncResult.Success -> it.value
-      }
-    }
+    return binding.root
   }
 
   private fun createLayoutManager(
-    recentStoryCount: Int,
-    oldStoryCount: Int,
-    suggestedStoryCount: Int
+    adapter: BindableAdapter<RecentlyPlayedItemViewModel>
   ): RecyclerView.LayoutManager {
-    val sectionTitle0Position = if (recentStoryCount == 0) {
-      // If recent story count is 0, that means that section title 0 will not be visible.
-      -1
-    } else {
-      0
-    }
-    val sectionTitle1Position = if (oldStoryCount == 0) {
-      // If old story count is 0, that means that section title 1 will not be visible.
-      -1
-    } else if (recentStoryCount == 0) {
-      0
-    } else {
-      recentStoryCount + 1
-    }
-    val sectionTitle2Position = when {
-      suggestedStoryCount == 0 -> {
-        -1 // If suggested story count is 0, that means that section title 1 will not be visible.
-      }
-      oldStoryCount == 0 && recentStoryCount == 0 -> {
-        0
-      }
-      oldStoryCount > 0 && recentStoryCount > 0 -> {
-        recentStoryCount + oldStoryCount + 2
-      }
-      else -> {
-        recentStoryCount + oldStoryCount + 1
-      }
-    }
 
     val spanCount = activity.resources.getInteger(R.integer.recently_played_span_count)
-    promotedStoryListAdapter.setSpanCount(spanCount)
```

Comment:
Just checking: was this not needed before with the SpanSizeLookup being provided?

Reply:
Not needed: the method just sets a private field that is not used internally. https://github.com/oppia/oppia-android/blob/3059cc7870c521e6b52b5740e6b1982dd9fe0a2d/app/src/main/java/org/oppia/android/app/home/recentlyplayed/PromotedStoryListAdapter.kt#L23

Reply:
Nice catch!